pkg/ebpf/events_derived.go

```diff
@@ -6,87 +6,102 @@ import (
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
-type deriveFn func(EventDefinition) error
+//deriveFn represents a function which takes an event and derives whether another event should result from it
+//
+//returns a derived (or empty) event, whether an event was derived, and an error if relevant
```

Comment:
You could improve this comment, it is not very clear. Suggestion:

```go
// deriveFn is a function prototype for a function that receives an event as
// argument and produces a new event, depending on the event type received. It
// returns the derived event, a bool indicating if there was indeed a derived
// event created, and an error/nil.
```

Reply:
I've rewritten to this:

```
// deriveFn is a function prototype for a function that receives an event as
// argument and may produce a new event if relevant.
// It returns the a derived or empty event, depending on succesful derivation,
// a bool indicating if an event was derived, and an error if one occured.
```